src/scripts/jefferson

```diff
@@ -318,21 +384,42 @@ def scan_fs(content, endianness, verbose=False):
                     inode.unpack(content_mv[0 + offset :])
 
                     if inode.ino in fs[JFFS2_NODETYPE_INODE]:
-                        if inode.version > fs[JFFS2_NODETYPE_INODE][inode.ino].version:
-                            fs[JFFS2_NODETYPE_INODE][inode.ino] = inode
+                        fs[JFFS2_NODETYPE_INODE][inode.ino].append(inode)
                     else:
-                        fs[JFFS2_NODETYPE_INODE][inode.ino] = inode
+                        fs[JFFS2_NODETYPE_INODE][inode.ino] = [inode]
                     if verbose:
                         print("0x%08X:" % (offset), inode)
-                elif unknown_node.nodetype == JFFS2_NODETYPE_CLEANMARKER:
-                    pass
-                elif unknown_node.nodetype == JFFS2_NODETYPE_PADDING:
-                    pass
                 elif unknown_node.nodetype == JFFS2_NODETYPE_SUMMARY:
-                    pass
+                    summary = Jffs2_raw_summary()
+                    summary.unpack(content_mv[0 + offset :])
+                    fs[JFFS2_NODETYPE_SUMMARY].append(summary)
+                    if verbose:
+                        print("0x%08X:" % (offset), summary)
                 elif unknown_node.nodetype == JFFS2_NODETYPE_XATTR:
-                    pass
+                    xattr = Jffs2_raw_xattr()
+                    xattr.unpack(content_mv[0 + offset :])
+
+                    if xattr.ino in fs[JFFS2_NODETYPE_XATTR]:
+                        if xattr.version > fs[JFFS2_NODETYPE_XATTR][xattr.ino].version:
+                            fs[JFFS2_NODETYPE_XATTR][xattr.ino] = xattr
+                    else:
+                        fs[JFFS2_NODETYPE_XATTR][xattr.ino] = xattr
+                    if verbose:
+                        print("0x%08X:" % (offset), xattr)
                 elif unknown_node.nodetype == JFFS2_NODETYPE_XREF:
+                    xref = Jffs2_raw_xref()
+                    xref.unpack(content_mv[0 + offset :])
+
+                    if xref.ino in fs[JFFS2_NODETYPE_XREF]:
+                        if xref.version > fs[JFFS2_NODETYPE_XREF][xref.ino].version:
+                            fs[JFFS2_NODETYPE_XREF][xref.ino] = xref
+                    else:
+                        fs[JFFS2_NODETYPE_XREF][xref.ino] = xref
+                    if verbose:
+                        print("0x%08X:" % (offset), xref)
```

Comment:
Not needed because those node types are not used to reconstruct the filesystem on disk. We parse structures that are never used afterwards.